opp_path_selection/src/path_selector.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2018 Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opp_path_selection/path_selector.h"
+#include "opp_path_selection/filter_impl.hpp"
+
+#include <geometry_msgs/Point.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <sensor_msgs/PointCloud2.h>
+
+#include <boost/optional.hpp>
+#include <boost/graph/dijkstra_shortest_paths.hpp>
+#include <boost/graph/breadth_first_search.hpp>
+#include <boost/graph/graph_utility.hpp>
+#include <boost/pending/indirect_cmp.hpp>
+#include <boost/range/irange.hpp>
+#include <eigen_conversions/eigen_msg.h>
+#include <pcl/common/centroid.h>
+#include <pcl/common/common.h>
+#include <pcl/features/normal_3d.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/search/kdtree.h>
+#include <pcl/segmentation/extract_polygonal_prism_data.h>
+#include <pcl/segmentation/region_growing.h>
+
+namespace opp_path_selection
+{
+int PathSelector::pointToLineSegmentDistance(const Eigen::Vector3d& segStart,
+                                             const Eigen::Vector3d& segEnd,
+                                             const Eigen::Vector3d& pt,
+                                             double& distance_to_seg,
+                                             double& distance_along_seg)
+{
+  int rtn = 1;
+  Eigen::Vector3d line = segEnd - segStart;
+  Eigen::Vector3d ptln = pt - segStart;
+  double seg_len = line.norm();
+  line.normalize();
+
+  distance_to_seg = line.cross(ptln).norm();
+  distance_along_seg = line.dot(ptln);
+  if (distance_along_seg < 0)
+  {
+    Eigen::Vector3d dv = segStart - pt;
+    distance_to_seg = dv.norm();
+    rtn = 0;
+  }
+  else if (distance_along_seg > seg_len)
+  {
+    Eigen::Vector3d dv = segEnd - pt;
+    distance_to_seg = dv.norm();
+    rtn = -1;
+  }
+  return rtn;
+}
+
+std::vector<int> PathSelector::findPointsAlongSegments(const shape_msgs::Mesh& input_mesh,
+                                                       const std::vector<Eigen::Vector3d>& points,
+                                                       const PathSelectorParameters& params)
+{
+  // Check size of selection points vector
+  if (points.size() < 2)
+  {
+    ROS_ERROR("Must have at least 2 points in Path");
+    return {};
+  }
+  // find id of every point from marker array, needed for Dijkstras
+  std::vector<int> point_seg_indices;
+  for (int i = 0; i < points.size(); i++)
+  {
+    double mind = 2000.0;
```

Comment:
Replace arbitrary values with maximum limits

```suggestion
    double mind = std::numeric_limits<double>::max();
```